keps/sig-node/4939-tls-in-grpc-probe/README.md

```diff
@@ -0,0 +1,901 @@
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+The new gRPC health probe enables developers to probe
+[gRPC health servers](https://github.com/grpc-ecosystem/grpc-health-probe)
+from the node.
+This allows them to stop using workarounds such as this
+[grpc-health-probe](https://github.com/grpc-ecosystem/grpc-health-probe)
+paired with `exec` probes.
+
+It allows natively running health checks on gRPC services
+without deploying additional binaries as well as other benefits
+outlined in [the announcement](https://kubernetes.io/blog/2022/05/13/grpc-probes-now-in-beta/).
+
+A limitation in the current implementation is that it only supports
+gRPC servers that do not leverage TLS connections.
+Even if they are not concerned about certificate verification for the health check,
+a connection cannot be established at all if the server is expecting TLS and the client
+is not.
+
+This enhancement aims to add configuration options to enable TLS and possibly
+disable certificate validation and/or provide a CA cert to validate with.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+We often deploy internal gRPC services on our cluster.
+These deployments provide internal services and it's simple to add
+the health server to them so they can be verified through a single interface.
+
+It's also worth noting we have and internal CA that signs certs for communicating
+with these servers and all of them use TLS.
+
+Currently, we are using the `exec` probe for readiness and liveness configured as:
+
+```
+liveness_probe {
+  exec {
+    command = [
+      "/bin/grpc_health_probe",
+      "-addr=:8443",
+      "-tls",
+      "-tls-no-verify",
+    ]
+  }
+}
+```
+
+We would really like to switch to the gRPC probes introduced in 1.24
+but are unable to do so since there is no way to configure it to use a TLS connection
+when reaching out to the health server.
+
+Instead we must continue to rely on the `exec` probe and cannot reap the benefits
+described in [the announcement](https://kubernetes.io/blog/2022/05/13/grpc-probes-now-in-beta/).
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+The primary goal is to support TLS connections when using the `grpc` probe
+either through providing a CA certificate from a secret
+or disabling certificate verification.
```

Comment:
we don't validate certificates with HTTPS probes, I don't feel we should add this complexity in this iteration and just make grpc probes the same as http probes

Reply:
:+1: that makes sense to me, just pushed up some fixups to update the KEP 09308e90